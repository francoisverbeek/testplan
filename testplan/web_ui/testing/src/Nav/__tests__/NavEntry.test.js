import React from 'react';
import {shallow} from 'enzyme';
import {StyleSheetTestUtils} from "aphrodite";
import {Badge} from 'reactstrap';

import NavEntry from '../NavEntry';

function defaultProps() {
  return {
    name: 'entry name',
    status: 'passed',
    type: 'testplan',
    caseCountPassed: 0,
    caseCountFailed: 0,
  };
}

describe('NavEntry', () => {
  let props;
  let mountedNavEntry;
  const renderNavEntry = () => {
    if (!mountedNavEntry) {
      mountedNavEntry = shallow(
        <NavEntry {...props} />
      );
    }
    return mountedNavEntry;
  };

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
    props = defaultProps();
    mountedNavEntry = undefined;
  });

  afterEach(() => {
    // Resume style injection once test is finished.
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('shallow renders without crashing', () => {
    renderNavEntry();
  });

  it('shallow renders the correct HTML structure', () => {
    const navEntry = renderNavEntry();
    expect(navEntry).toMatchSnapshot();
  });

  it('when prop status="failed" name div and Badge have correct styles', () => {
    props.status = 'failed';
    const navEntry = renderNavEntry();

    const name = navEntry.children().first();
    expect(name).toHaveLength(1);
    const expectedName = defaultProps().name;
    expect(name.text()).toEqual(expectedName);
    expect(name.props().className).toMatch(/failed/);

    const badge = navEntry.find(Badge);
    expect(badge).toHaveLength(1);
    expect(badge.props().className).toMatch(/failedBadge/);
  });
});