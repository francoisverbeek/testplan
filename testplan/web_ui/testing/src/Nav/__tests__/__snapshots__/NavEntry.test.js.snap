// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavEntry shallow renders the correct HTML structure 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavEntry
    caseCountFailed={0}
    caseCountPassed={0}
    name="entry name"
    status="passed"
    type="testplan"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="entryName_1w2kiva-o_O-passed_1kh4m98"
>
          entry name
</div>,
        <div
          className="entryIcons_dx1noq"
>
          <i
                    className="entryIcon_ull2p3"
                    title="passed/failed testcases"
          >
                    <span
                              className="passed_1kh4m98"
                    />
                    /
                    <span
                              className="failed_11es5k7"
                    />
          </i>
          <Badge
                    className="entryIcon_ull2p3-o_O-passedBadge_1gkx146"
                    color="secondary"
                    pill={true}
                    tag="span"
                    title="testplan"
          >
                    TP
          </Badge>
</div>,
      ],
      "className": "d-flex justify-content-between",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "entry name",
          "className": "entryName_1w2kiva-o_O-passed_1kh4m98",
        },
        "ref": null,
        "rendered": "entry name",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              className="entryIcon_ull2p3"
              title="passed/failed testcases"
>
              <span
                            className="passed_1kh4m98"
              />
              /
              <span
                            className="failed_11es5k7"
              />
</i>,
            <Badge
              className="entryIcon_ull2p3-o_O-passedBadge_1gkx146"
              color="secondary"
              pill={true}
              tag="span"
              title="testplan"
>
              TP
</Badge>,
          ],
          "className": "entryIcons_dx1noq",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="passed_1kh4m98"
/>,
                "/",
                <span
                  className="failed_11es5k7"
/>,
              ],
              "className": "entryIcon_ull2p3",
              "title": "passed/failed testcases",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 0,
                  "className": "passed_1kh4m98",
                },
                "ref": null,
                "rendered": 0,
                "type": "span",
              },
              "/",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 0,
                  "className": "failed_11es5k7",
                },
                "ref": null,
                "rendered": 0,
                "type": "span",
              },
            ],
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "TP",
              "className": "entryIcon_ull2p3-o_O-passedBadge_1gkx146",
              "color": "secondary",
              "pill": true,
              "tag": "span",
              "title": "testplan",
            },
            "ref": null,
            "rendered": "TP",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="entryName_1w2kiva-o_O-passed_1kh4m98"
>
            entry name
</div>,
          <div
            className="entryIcons_dx1noq"
>
            <i
                        className="entryIcon_ull2p3"
                        title="passed/failed testcases"
            >
                        <span
                                    className="passed_1kh4m98"
                        />
                        /
                        <span
                                    className="failed_11es5k7"
                        />
            </i>
            <Badge
                        className="entryIcon_ull2p3-o_O-passedBadge_1gkx146"
                        color="secondary"
                        pill={true}
                        tag="span"
                        title="testplan"
            >
                        TP
            </Badge>
</div>,
        ],
        "className": "d-flex justify-content-between",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "entry name",
            "className": "entryName_1w2kiva-o_O-passed_1kh4m98",
          },
          "ref": null,
          "rendered": "entry name",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="entryIcon_ull2p3"
                title="passed/failed testcases"
>
                <span
                                className="passed_1kh4m98"
                />
                /
                <span
                                className="failed_11es5k7"
                />
</i>,
              <Badge
                className="entryIcon_ull2p3-o_O-passedBadge_1gkx146"
                color="secondary"
                pill={true}
                tag="span"
                title="testplan"
>
                TP
</Badge>,
            ],
            "className": "entryIcons_dx1noq",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="passed_1kh4m98"
/>,
                  "/",
                  <span
                    className="failed_11es5k7"
/>,
                ],
                "className": "entryIcon_ull2p3",
                "title": "passed/failed testcases",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 0,
                    "className": "passed_1kh4m98",
                  },
                  "ref": null,
                  "rendered": 0,
                  "type": "span",
                },
                "/",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 0,
                    "className": "failed_11es5k7",
                  },
                  "ref": null,
                  "rendered": 0,
                  "type": "span",
                },
              ],
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "TP",
                "className": "entryIcon_ull2p3-o_O-passedBadge_1gkx146",
                "color": "secondary",
                "pill": true,
                "tag": "span",
                "title": "testplan",
              },
              "ref": null,
              "rendered": "TP",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
